File: backend/app/schemas/csv_imports.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.manufacturing import CsvImportStatus, CsvImportType


class CsvImportErrorResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    row_number: int
    field_name: str
    error_code: str
    error_message: str
    input_value: str


class CsvImportJobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    import_type: CsvImportType
    file_name: str
    status: CsvImportStatus
    total_rows: int
    success_rows: int
    error_rows: int
    accepted_at: datetime
    completed_at: datetime | None
    errors: list[CsvImportErrorResponse]
    error_csv_available: bool

File: frontend/src/api/csvImports.ts
import { apiFetch, apiUrl } from "./client";

export type CsvImportErrorItem = {
  row_number: number;
  field_name: string;
  error_code: string;
  error_message: string;
  input_value: string;
};

export type CsvImportJob = {
  id: number;
  import_type: "PRODUCT_MASTER";
  file_name: string;
  status: "PROCESSING" | "SUCCEEDED" | "FAILED";
  total_rows: number;
  success_rows: number;
  error_rows: number;
  accepted_at: string;
  completed_at: string | null;
  errors: CsvImportErrorItem[];
  error_csv_available: boolean;
};

export function uploadProductsCsv(file: File): Promise<CsvImportJob> {
  const body = new FormData();
  body.append("file", file);
  return apiFetch<CsvImportJob>("/imports/products", { method: "POST", body });
}

export function productImportErrorCsvUrl(jobId: number): string {
  return apiUrl(`/imports/products/${jobId}/errors.csv`);
}
